boost causes a linker error when building ledger

Is this maybe due to it being built statically? Why _is_ it built statically in the first place? @cemkeylan 

```
/usr/lib/gcc/x86_64-pc-linux-musl/10.0.1/../../../../x86_64-pc-linux-musl/bin/ld: /usr/lib64/libboost_filesystem.a(directory.o): relocation R_X86_64_PC32 against symbol `_ZN5boost6system6detail10cat_holderIvE24system_category_instanceE' can not be used when making a shared object; recompile with -fPIC
/usr/lib/gcc/x86_64-pc-linux-musl/10.0.1/../../../../x86_64-pc-linux-musl/bin/ld: final link failed: bad value
collect2: error: ld returned 1 exit status
ninja: subcommand failed
```

--- a/community/boost/build
+++ b/community/boost/build
@@ -1,5 +1,5 @@
 #!/bin/sh -e
 
 ./bootstrap.sh --prefix="$1/usr"
-./b2 stage threading=multi link=shared
-./b2 install threading=multi link=shared
+./b2 stage threading=multi
+./b2 install threading=multi

--- a/community/boost/build
+++ b/community/boost/build
@@ -1,5 +1,5 @@
 #!/bin/sh -e
 
 ./bootstrap.sh --prefix="$1/usr"
-./b2 stage threading=multi link=static
-./b2 install threading=multi link=static
+./b2 stage threading=multi link=shared
+./b2 install threading=multi link=shared

